cmd/prometheus/main.go

```diff
@@ -757,3 +780,24 @@ func sendAlerts(s sender, externalURL string) rules.NotifyFunc {
 		}
 	}
 }
+
+// chooseRetention is some roundabout code to support both RetentionDuration and Retention (for different flags).
+// If Retention is 15d, then it means that the default value is set and the value of RetentionDuration is used.
+func chooseRetention(oldFlagDuration, newFlagDuration model.Duration) model.Duration {
+	defaultDuration, err := model.ParseDuration(defaultRetentionString)
+	if err != nil {
+		panic(err)
+	}
+
+	retention := oldFlagDuration
+	if retention == defaultDuration {
+		retention = newFlagDuration
+	}
+
+	// Further if both the flags are set, then RetentionDuration takes precedence.
+	if newFlagDuration != defaultDuration {
```

Comment:
I'm not sure this yields the expected behavior, when we say the new flag takes precedence. If the new flag explicitly specifies 15 days and the old flag specifies 30 days, then the old one will be chosen.

Reply:
Hmm, fair. Not sure how to deal with the case though :/ Detecting if the flag has been set or if it's just the default value is next to impossible.

Reply:
I think just setting the flag description to explain this rectifies the situation.

"If this flag is set to anything other than the default, it takes precedence."

Reply:
Done